src/link/Elf.zig

```diff
@@ -1767,7 +1770,11 @@ fn linkWithLLD(self: *Elf, comp: *Compilation, prog_node: *std.Progress.Node) !v
                 try argv.append("-no-whole-archive");
                 whole_archive = false;
             }
-            try argv.append(obj.path);
+
+            if (obj.shlib_path) |shlib_path| {
```

Comment:
If `is_shlib: bool` was available, we could simply recreate the input from `path` as follows:

```zig
if (obj.is_shlib) {
    try argv.append("-l");
    try argv.append(try std.fmt.fmtPrintAlloc(arena, ":{s}", .{obj.path});
}
```